Source/WebCore/css/CSSPrimitiveValue.h

```diff
@@ -155,42 +157,66 @@ class CSSPrimitiveValue final : public CSSValue {
     ~CSSPrimitiveValue();
 
     CSSUnitType primitiveType() const;
-    ExceptionOr<float> getFloatValue(CSSUnitType) const;
-
-    double computeDegrees() const;
-    template<typename T, TimeUnit> T computeTime() const;
-    template<typename T, FrequencyUnit> T computeFrequency() const;
-
-    template<typename T> T computeLength(const CSSToLengthConversionData&) const;
-    template<int> Length convertToLength(const CSSToLengthConversionData&) const;
 
+    // Exposed for DeprecatedCSSOMPrimitiveValue. Throws if conversion to `targetUnit` is not allowed.
+    ExceptionOr<float> getFloatValueDeprecated(CSSUnitType targetUnit) const;
+
+    // MARK: Integer (requires `isInteger() == true`)
+    template<typename T = int> T resolveInteger(const CSSToLengthConversionData&) const;
+    template<typename T = int> T resolveIntegerNoConversionDataRequired() const;
+    template<typename T = int> T resolveIntegerDeprecated() const;
+
+    // MARK: Number (requires `isNumberOrInteger() == true`)
```

Comment:
Unclear that `isNumberOrInteger` corresponds to `AsNumber`. The latter sounds like it could do other conversions.

Reply:
In reality, "integer" is really just a "number" that we know doesn't have a fractional part. They are both stored exactly the same. I am not sure there are really any useful places where `isNumber()` would the right check, and `isNumberOrInteger()` should really always be used.